### PR TITLE
Replace unwrap() with expect() or proper handling

### LINKED ISSUE #56: Our startup error messages / handling are horrible
If you forget to specify a required parameter, all you get is `thread 'main' panicked at 'called`Option::unwrap()`on a`None`value', ../src/libcore/option.rs:323`

We should log something sane, here.

### PR DESCRIPTION
**Pull request by callahad**
Still not perfect -- we can still panic in a bunch of places -- but at least this ensures we leave behind sane error messages.

Fixes #56

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -48,19 +49,27 @@ struct Args {
...
                          .unwrap_or_else(|e| e.exit());
 
     // Read the configuration from the provided file.
     let mut builder = broker::config::ConfigBuilder::new();
     if let Some(path) = args.arg_CONFIG {
-        builder.update_from_file(&path).unwrap();
+        match builder.update_from_file(&path) {
+            Ok(_) => (),
+            Err(err) => panic!(format!("failed to read config file: {}", err.description())),
+        }
```
Comment (stephank) [inline src/main.rs]: Could also do this with `if let Err(err)` or `unwrap_or_else`?

File: src/main.rs [hunk 2]
```diff
@@ -48,19 +49,27 @@ struct Args {
...
     }
     builder.update_from_common_env();
     builder.update_from_broker_env();
-    let app = Arc::new(builder.done().unwrap());
+    let app = Arc::new(match builder.done() {
+        Ok(config) => config,
+        Err(err) => panic!(format!("failed to realize configuration: {}", err.description())),
+    });
```
Comment (stephank) [inline src/main.rs]: `unwrap_or_else`?

File: src/oidc_bridge.rs [hunk 3]
```diff
@@ -137,7 +137,10 @@ pub fn canonicalized(email: &str) -> String {
 pub fn verify(app: &Config, stored: &HashMap<String, String>, code: &str)
               -> BrokerResult<(String, String)> {
 
-    let email_addr = EmailAddress::new(&stored["email"]).unwrap();
+    let email_addr = match EmailAddress::new(&stored["email"]) {
+        Err(_) => return Err(BrokerError::Custom("Error parsing stored email address".to_string())),
+        Ok(addr) => addr,
+    };
```
Comment (stephank) [inline src/oidc_bridge.rs]: `unwrap_or_else`?

File: src/crypto.rs [hunk 4]
```diff
@@ -88,9 +91,10 @@ impl NamedKey {
                 .insert("kid", &self.id)
                 .insert("alg", "RS256")
                 .build()
-            ).unwrap();
+            ).expect("unable to coerce jwt header to string");
```
Comment (stephank) [inline src/crypto.rs]: coerce *into*

File: src/handlers/mod.rs [hunk 5]
```diff
@@ -166,7 +166,7 @@ fn handle_error(app: &Config, req: &mut Request, err: BrokerError) -> IronResult
 /// Serializes the argument value to JSON and returns a HTTP 200 response
 /// code with the serialized JSON as the body.
 fn json_response(obj: &Value) -> BrokerResult<Response> {
-    let content = serde_json::to_string(&obj).unwrap();
+    let content = serde_json::to_string(&obj).expect("unable to coerce JSON Value to string");
```
Comment (stephank) [inline src/handlers/mod.rs]: coerce *into*

File: src/oidc_bridge.rs (hunk 3 above)
Comment (callahad) [inline src/oidc_bridge.rs]: Even better, `?` :)

**Pull request merged**